components/feature/tabs/src/main/java/mozilla/components/feature/tabs/tabstray/TabsTrayPresenter.kt

```diff
@@ -31,14 +31,21 @@ class TabsTrayPresenter(
     private val closeTabsTray: () -> Unit
 ) {
     private var tabs: Tabs? = null
-    private var scope: CoroutineScope? = null
+    private var mediaScope: CoroutineScope? = null
+    private var tabScope: CoroutineScope? = null
 
     fun start() {
-        scope = store.flowScoped { flow -> collect(flow) }
+        tabScope = store.flowScoped { flow -> collect(flow) }
+        mediaScope = store.flowScoped {
+            it
+                .ifChanged { state -> state.media.aggregate }
+                .collect { this.tabs?.also(tabsTray::updateTabs) }
```

Comment:
This evades the `tabFilter`, see above comment on how we can fix this and the previous concern.

Reply:
@jonalmeida We're not taking any tabs off of browserState here. `this.tabs` was set with the `tabFilter`